### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by pkerpedjiev**
## Description

* Added documentation for how to aggregate bed files as bedpe files
* Added a section on the POST API for tilesets
* Rearranged the server sections so they have a (slightly) more logical ordering.

Fixes #___

## Checklist

- [ ] Unit tests added or updated
- [x] Documentation added or updated
- [ ] Example added or updated
- [ ] Screenshot for visual changes (e.g. new tracks)
- [ ] Updated CHANGELOG.md

### CODE REVIEW

File: docs/data_preparation.rst [hunk 1]
```diff
@@ -98,15 +98,35 @@ contain too many values and slow down the renderer:
         --output-file domains.txt.multires \
         domains.txt
 
-This requires the `--chr1-col`, `--from1-col`, `--to1-col`, `--chr2-col`,
-`--from2-col`, `--to2-col` parameters to specify which columns in the datafile
+This requires the ``-chr1-col``, ``--from1-col``, ``--to1-col``, ``--chr2-col``,
```
Comment (mccalluc) [inline docs/data_preparation.rst]: `-chr1-col` -> `--chr1-col`

File: docs/data_preparation.rst [hunk 2]
```diff
@@ -98,15 +98,35 @@ contain too many values and slow down the renderer:
...
 
 The priority with which regions are included in lower resolution tiles is
-specified by the `--impotance-column` parameter. This can either provide a
-value, contain `random`, or if it's not specified, default to the size of the
+specified by the ``--impotance-column`` parameter. This can either provide a
+value, contain ``random``, or if it's not specified, default to the size of the
 region.
 
+**BED files** can also be aggregated as BEDPE-like files for use with the
+`2d-rectangle-domains` track. The from1_col,to1_col and from2_col,to2_col
```
Comment (mccalluc) [inline docs/data_preparation.rst]: More back-ticks.

File: docs/data_preparation.rst [hunk 3]
```diff
@@ -98,15 +98,35 @@ contain too many values and slow down the renderer:
...
+parameters need to be set to the same columns. Example file:
+
+.. code-block:: bash
```
Comment (mccalluc) [inline docs/data_preparation.rst]: I think just indenting is sufficient for example data.

File: docs/higlass_server.rst [hunk 4]
```diff
@@ -65,124 +65,75 @@ Additionally, the following settings are set via ``config.json`` (see ``config.j
...
 
-    docker exec higlass-container python \
-            higlass-server/manage.py ingest_tileset \
-            --filename /tmp/chromsizes_hg19.tsv \
-            --filetype chromsizes-tsv \
-            --datatype chromsizes
+  curl localhost:8000/api/v1/tilesets?dt=matrix
 
-Or using curl:
+Use the `dt` parameter to retrieve tilesets for a specfific datatype:
```
Comment (mccalluc) [inline docs/higlass_server.rst]: The examples are good, but the text seems redundant? 

> To filter by datatype, use the `dt=datatype` parameter:

vs

>  Use the `dt` parameter to retrieve tilesets for a specfific datatype:

File: docs/higlass_server.rst [hunk 5]
```diff
@@ -65,124 +65,75 @@ Additionally, the following settings are set via ``config.json`` (see ``config.j
...
 
-.. code-block:: bash
-
-    curl localhost:8000/api/v1/tilesets?dt=gene-annotation
-    curl localhost:8000/api/v1/tilesets?dt=chromsizes
-
-Retrieving properties of a tileset, for a specific `uuid`:
+The server API can be used to upload entire tilesets. To use this
+functionality, you need a username and password. These can be created using
+``higlass-manage create superuser``. The should be entered into a file (e.g.
```
Comment (mccalluc) [inline docs/higlass_server.rst]: "The" -> "They"

Comment (mccalluc) [reply docs/higlass_server.rst]: Could you say:

```
higlass-manage create superuser > ~/.higlass-server-login
```

File: docs/higlass_server.rst [hunk 6]
```diff
@@ -65,124 +65,75 @@ Additionally, the following settings are set via ``config.json`` (see ``config.j
...
- .. code-block:: bash
+.. code-block:: bash
 
-    curl --user ${username}:${password} --request PATCH --header "Content-Type: application/json" --data '{"name":"new_name_of_tileset"}' http://localhost:8000/api/v1/tilesets/${uuid}/
+  curl -u `cat ~/.higlass-server-login` \
+      -F "datafile=@/Users/peter/projects/negspy/negspy/data/mm10/chromInfo.txt" \
+      -F "filetype=chromsizes-tsv" \
+      -F "datatype=chromsizes" \
+      -F "coordSystem=mm10" \
+      -F "name=Chromosomes (mm10)" \
```
Comment (mccalluc) [inline docs/higlass_server.rst]: FWIW: I would use single-quotes unless we know that we need interpolation. Bash can be "helpful" when you're not expecting it.

File: docs/higlass_server.rst [hunk 7]
```diff
@@ -246,8 +197,65 @@ To modify the name of a tileset:
...
+
+    curl -u `cat ~/.higlass-server-login` \
+        -F "datafile=@/Users/peter/projects/negspy/negspy/data/mm10/chromInfo.txt" \
+        -F "filetype=chromsizes-tsv" \
+        -F "datatype=chromsizes" \
+        -F "coordSystem=mm10" \
+        -F "name=Chromosomes (mm10)" \
+        http://higlass.io/api/v1/tilesets/
+
+This should return a JSON object contain a UUID to confirm that the data has been
```
Comment (mccalluc) [inline docs/higlass_server.rst]: "contain" -> "containing"

**Comment by pkerpedjiev** [comment]
Thanks Chuck! Hoping to get `v1.5.8` out tomorrow.

**Pull request merged**